### PR TITLE
feat: add stopOnHover to AutoPlay

### PR DESCRIPTION
**Pull request by WoodNeck**
## Issue
#10 

## Details
This adds `stopOnHover` option for AutoPlay.

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export function merge(target: object, ...srcs: object[]): object {
```
Comment (mixed) [inline src/utils.ts:1]: Why not use `Object.assign` because you need to use polyfill?

Comment (WoodNeck) [reply src/utils.ts:1]: Sure, that's the reason. As we're targeting IE9+

File: src/AutoPlay.ts [hunk 2]
```diff
@@ -1,19 +1,58 @@
...
+  constructor(options: Partial<AutoPlayOptions> = DEFAULT_OPTION, _direction: AutoPlayOptions["direction"] = DEFAULT_OPTION.direction) {
+    if (typeof options === "number") {
+      // Fallback for previous interface
+      this.duration = options as number;
+      this.direction = _direction;
+      this.stopOnHover = DEFAULT_OPTION.stopOnHover;
+      return;
+    }
+
+    const mergedOptions = merge({}, DEFAULT_OPTION, options) as AutoPlayOptions;
```
Comment (daybrush) [inline src/AutoPlay.ts:48]: @WoodNeck 
it's like
```js
{
    ... DEFAULT_OPTION,
   ...options
}
```

File: src/utils.ts (hunk 1 above)
Comment (daybrush) [reply src/utils.ts:1]: @WoodNeck 

Create object.assign polyfill in tsc.

File: src/AutoPlay.ts [hunk 3]
```diff
@@ -1,19 +1,58 @@
...
-   * @param - The direction in which the panel moves. <ko>패널이 움직이는 방향</ko>
+   * @param options Options for the AutoPlay instance.<ko>AutoPlay 옵션</ko>
+   * @param options.duration Time to wait before moving on to the next panel.<ko>다음 패널로 움직이기까지 대기 시간</ko>
+   * @param options.direction The direction in which the panel moves.<ko>패널이 움직이는 방향</ko>
+   * @param options.stopOnHover Whether to stop when mouse hover upon the element.<ko>엘리먼트에 마우스를 올렸을 때 AutoPlay를 정지할지 여부</ko>
    * @example
    * flicking.addPlugins(new eg.Flicking.plugins.AutoPlay(2000, "NEXT"));
    */
-  constructor(private duration: number = 2000, private direction: Direction[keyof Direction] = "NEXT") {}
+  constructor(options: Partial<AutoPlayOptions> = DEFAULT_OPTION, _direction: AutoPlayOptions["direction"] = DEFAULT_OPTION.direction) {
```
Comment (daybrush) [inline src/AutoPlay.ts:39]: @WoodNeck 
Shouldn't we use underbar?

File: src/AutoPlay.ts (hunk 2 above)
Comment (WoodNeck) [reply src/AutoPlay.ts:48]: That works, it looks like we have to configure Flicking also.

File: src/AutoPlay.ts (hunk 3 above)
Comment (WoodNeck) [reply src/AutoPlay.ts:39]: Oops, maybe my linter was broken. I've fixed it.

**Pull request merged**